fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/LuceneIndexTest.java

```diff
@@ -2528,162 +3119,310 @@ private String numbersText(int i) {
                 Arrays.stream(words).map(word -> "+\"" + word + "\"").collect(Collectors.joining(" AND ")) +
                 ")";
     }
 
-    @Test
-    void scanWithQueryOnlySynonymIndex() {
+    @ParameterizedTest
+    @MethodSource(LUCENE_INDEX_MAP_PARAMS)
+    void scanWithQueryOnlySynonymIndex(Map<String, Index> indexMap, boolean isSynthetic) {
+        final Index index = indexMap.get(QUERY_ONLY_SYNONYM_LUCENE_INDEX_KEY);
         try (FDBRecordContext context = openContext()) {
-            rebuildIndexMetaData(context, SIMPLE_DOC, QUERY_ONLY_SYNONYM_LUCENE_INDEX);
-            recordStore.saveRecord(createSimpleDocument(1623L, "Hello record layer", 1));
-            assertIndexEntryPrimaryKeys(Set.of(1623L),
-                    recordStore.scanIndex(QUERY_ONLY_SYNONYM_LUCENE_INDEX, fullTextSearch(QUERY_ONLY_SYNONYM_LUCENE_INDEX, "hullo record layer"), null, ScanProperties.FORWARD_SCAN));
-            recordStore.saveRecord(createSimpleDocument(1623L, "Hello recor layer", 1));
-            assertIndexEntryPrimaryKeys(Set.of(1623L),
-                    recordStore.scanIndex(QUERY_ONLY_SYNONYM_LUCENE_INDEX, fullTextSearch(QUERY_ONLY_SYNONYM_LUCENE_INDEX, "hullo record layer"), null, ScanProperties.FORWARD_SCAN));
-            // "hullo" is synonym of "hello"
-            recordStore.saveRecord(createSimpleDocument(1623L, "Hello record layer", 1));
-            assertIndexEntryPrimaryKeys(Set.of(1623L),
-                    recordStore.scanIndex(QUERY_ONLY_SYNONYM_LUCENE_INDEX, fullTextSearch(QUERY_ONLY_SYNONYM_LUCENE_INDEX, matchAll("hullo", "record", "layer")), null, ScanProperties.FORWARD_SCAN));
-            // it doesn't match due to the "recor"
-            recordStore.saveRecord(createSimpleDocument(1623L, "Hello record layer", 1));
-            assertIndexEntryPrimaryKeys(Collections.emptySet(),
-                    recordStore.scanIndex(QUERY_ONLY_SYNONYM_LUCENE_INDEX, fullTextSearch(QUERY_ONLY_SYNONYM_LUCENE_INDEX, matchAll("hullo", "recor", "layer")), null, ScanProperties.FORWARD_SCAN));
-            // "hullo" is synonym of "hello", and "show" is synonym of "record"
-            recordStore.saveRecord(createSimpleDocument(1623L, "Hello record layer", 1));
-            assertIndexEntryPrimaryKeys(Set.of(1623L),
-                    recordStore.scanIndex(QUERY_ONLY_SYNONYM_LUCENE_INDEX, fullTextSearch(QUERY_ONLY_SYNONYM_LUCENE_INDEX, matchAll("hullo", "show", "layer")), null, ScanProperties.FORWARD_SCAN));
+            if (isSynthetic) {
+                openRecordStore(context, metaDataBuilder -> metaDataHookSyntheticRecord1(metaDataBuilder, index));
+                Tuple primaryKey = createSyntheticRecord1(1, 1623L, 1623L, "Hello record layer", "", false, System.currentTimeMillis(), 0);
+                assertIndexEntryPrimaryKeyTuples(Set.of(primaryKey),
+                        recordStore.scanIndex(index, fullTextSearch(index, "simple_text:hullo simple_text:record simple_text:layer"), null, ScanProperties.FORWARD_SCAN));
```

Comment:
when specifying a field, the field needs to be applied to every token (since this is not an exact phrase search). 
I tried to cover both cases: search with a field specified and search all fields with synthetic record.